controllers/clusterurlmonitor/clusterurlmonitor_supplement.go

```diff
@@ -67,6 +68,16 @@ func (s *ClusterUrlMonitorReconciler) EnsurePrometheusRuleExists(clusterUrlMonit
 	return utilreconcile.ContinueReconcile()
 }
 
+// FindStatusCondition finds the conditionType in conditions.
+func (s *ClusterUrlMonitorReconciler) FindStatusCondition(conditions []metav1.Condition, conditionType string) *metav1.Condition {
```

Comment:
We can simplify this logic by having one helper function determine if cluster is available (eg. https://github.com/openshift/hypershift/pull/2147/files#diff-3d4ffda0834e09a59a4d510523932e20b9234bcb9816e40258a0fb32a87a7006R187) then requeue when !isClusterVersionAvailable